File: frontend/services/phone.ts
export function normalizePhone(phone: string): string {
  return phone.trim().replace("+91", "").replace(/\s+/g, "").replace(/-/g, "");
}

export function formatPhoneForInput(phone: string): string {
  const digits = normalizePhone(phone).replace(/\D/g, "").slice(0, 10);
  if (digits.length <= 5) {
    return digits;
  }
  return `${digits.slice(0, 5)} ${digits.slice(5)}`;
}

export function phoneCandidates(phone: string): string[] {
  const raw = phone.trim();
  const normalized = normalizePhone(phone);
  const compact = raw.replace(/\s+/g, "").replace(/-/g, "");
  return Array.from(new Set([normalized, raw, compact]));
}

File: backend/utils/phone.py
def normalize_phone(phone: str) -> str:
    return phone.strip().replace("+91", "").replace(" ", "").replace("-", "")


def phone_candidates(phone: str) -> list[str]:
    raw = phone.strip()
    normalized = normalize_phone(phone)
    candidates = [normalized, raw, raw.replace(" ", "").replace("-", "")]
    return list(dict.fromkeys(candidates))
